mne_connectivity/datasets/surrogate.py

```diff
@@ -0,0 +1,141 @@
+        specified.
+    return_generator : bool (default True)
+        Whether or not to return the surrogate data as a :term:`generator` object
+        instead of a :class:`list`. This allows iterating over the surrogates without
+        having to keep them all in memory.
+
+    Returns
+    -------
+    surrogate_data : list of ~mne.time_frequency.EpochsSpectrum
+        The surrogate data for the null hypothesis with ``n_shuffles`` entries. Returned
+        as a :term:`generator` if ``return_generator=True``.
+
+    Notes
+    -----
+    Surrogate data is generated by randomly shuffling the order of epochs, independently
+    for each channel. This destroys the covariance of the data, such that connectivity
+    estimates should reflect the null hypothesis of no genuine connectivity between
+    signals (e.g., only interactions due to background noise)
+    :footcite:`PellegriniEtAl2023`.
+
+    For the surrogate data to properly reflect a null hypothesis, the data which is
+    shuffled **must not** have a temporal structure that is consistent across epochs.
+    Examples of this data include evoked potentials, where a stimulus is presented or an
+    action performed at a set time during each epoch. Such data should not be used for
+    generating surrogates, as even after shuffling the epochs, it will still show a high
+    degree of residual connectivity between channels. As a result, connectivity
+    estimates from your surrogate data will capture genuine interactions, instead of the
+    desired background noise. Treating these estimates as a null hypothesis will
+    increase the likelihood of a type II (false negative) error, i.e., that there is no
+    significant connectivity in your data.
+
+    Appropriate data for generating surrogates includes data from a resting state,
+    inter-trial period, or similar. Here, a strong temporal consistency across epochs is
+    not assumed, reducing the chances that connectivity information of interest is
+    captured in your surrogate connectivity estimates.
+
+    In situations where you want to assess whether evoked data has significant
+    connectivity, you can generate your surrogate connectivity estimates from non-evoked
+    data (e.g., rest data, inter-trial data) and compare this to your true connectivity
+    estimates from the evoked data.
+
+    Regardless of whether you are working with evoked or non-evoked data, **you should
+    always compare true and surrogate connectivity estimates from epochs of the same
+    duration**. This will ensure that spectral information is captured with the same
+    accuracy in both sets of connectivity estimates. Ideally, **you should also compare
+    true and surrogate connectivity estimates from the same number of epochs** to avoid
+    biases from noise (fewer epochs gives noisier estimates) or finite sample sizes
+    (e.g., in coherency, phase-locking value, etc... :footcite:`VinckEtAl2010`).
```

Comment:
Very long notes section, but like we discussed, there are important considerations as to what data surrogates should be generated from.